core/dbt/task/debug.py

```diff
@@ -310,72 +437,57 @@ def test_configuration(self):
                 )
             )
 
-        self._log_project_fail()
-        self._log_profile_fail()
-
-    def _log_project_fail(self):
-        if not self.project_fail_details:
-            return
-
-        self.any_failure = True
-        if self.project_fail_details == FILE_NOT_FOUND:
-            return
-        msg = (
-            f"Project loading failed for the following reason:"
-            f"\n{self.project_fail_details}"
-            f"\n"
-        )
-        self.messages.append(msg)
-
-    def _log_profile_fail(self):
-        if not self.profile_fail_details:
-            return
-
-        self.any_failure = True
-        if self.profile_fail_details == FILE_NOT_FOUND:
-            return
-        msg = (
-            f"Profile loading failed for the following reason:"
-            f"\n{self.profile_fail_details}"
-            f"\n"
-        )
-        self.messages.append(msg)
+    # ===============
+    # Connection test
+    # ===============
 
     @staticmethod
-    def attempt_connection(profile):
-        """Return a string containing the error message, or None if there was
-        no error.
-        """
+    def attempt_connection(profile) -> Optional[str]:
+        """Return a string containing the error message, or None if there was no error."""
         register_adapter(profile)
         adapter = get_adapter(profile)
         try:
             with adapter.connection_named("debug"):
+                # set in adapter repos
```

Comment:
super nit: would `# implemented in adapter class` be more accurate? (technically postgres is in the same repo after all

Reply:
You're so right lol. On it